test/e2e/egress_test.go

```diff
@@ -584,21 +584,17 @@ func testEgressNodeFailure(t *testing.T, data *TestData) {
 			case 6:
 				skipIfNotIPv6Cluster(t)
 			}
-			signalAgent := func(nodeName, signal string) {
-				cmd := fmt.Sprintf("pkill -%s antrea-agent", signal)
-				rc, stdout, stderr, err := RunCommandOnNode(nodeName, cmd)
-				if rc != 0 || err != nil {
-					t.Errorf("Error when running command '%s' on Node '%s', rc: %d, stdout: %s, stderr: %s, error: %v",
-						cmd, nodeName, rc, stdout, stderr, err)
-				}
-			}
-			pauseAgent := func(_, evictNode string) {
+			pauseAgent := func(t *testing.T, _, evictNode string) {
 				// Send "STOP" signal to antrea-agent.
-				signalAgent(evictNode, "STOP")
+				if err := data.signalAntreaAgent(evictNode, "STOP"); err != nil {
+					t.Skipf("Skipping test because failed to pause antrea-agent (lack of permission?): %v", err)
```

Comment:
`err` should already have the reason so need `(lack of permission?)`?
However, if the current two ways of sending signal can work for all CI testbeds, do we still need to skip it when error happens? For local test, it should be not hard to get sudo permission if the error is clear.

Reply:
Do we have coverage enabled on the Jenkins testbeds? I saw the error as follows:
```
=== RUN   TestEgress/testEgressNodeFailure/IPv6_cluster
    egress_test.go:590: Skipping test because failed to pause antrea-agent (lack of permission?): Error when running command "pkill -STOP antrea-agent" on Node "antrea-ipv6-2-0", rc: 0, stdout: , stderr: pkill: killing pid 22488 failed: Operation not permitted
        , error: <nil>
```
How about we patch the agent container start command with a wrapper `sh -c` to work around the pid issue for this test case?

Reply:
sounds good to me. I suppose you mean updating the daemonset when running TestEgress if it's not already wrapped.